unstructured/partition/lang.py

```diff
@@ -17,3 +156,55 @@ def convert_old_ocr_languages_to_languages(ocr_languages: str):
     """
 
     return ocr_languages.split("+")
+
+
+@requires_dependencies("pytesseract")
+def convert_language_to_tesseract(lang: str) -> str:
+    """
+    Convert a language code to its tesseract formatted and recognized langcode(s), if supported.
+    """
+    # if language is already tesseract langcode, return it immediately
+    # this will catch the tesseract special cases equ and osd
+    # NOTE(shreya): this may catch some of the cases of choosing between a plain vs suffixed tesseract code
+    if lang in PYTESSERACT_LANGS:
+        return lang
+
+    # get iso639 language object
+    try:
+        lang_iso639 = iso639.Language.match(lang)
+    except:
+        logger.warning(f"{lang} is not a valid standard language code.")
+        return ""
+
+    # tesseract uses 3 digit codes as prefix, with suffixes for orthography
+    # match to first 3 letters of tesseract codes
+    pytesseract_langs_3 = [lang[:3] for lang in PYTESSERACT_LANGS]
+
+    # try to match 639-3 (part3)
+    if lang_iso639.part3 in pytesseract_langs_3:
+        print("match in part3")
+        matched_langcodes = _get_all_tesseract_langcodes_with_prefix(lang_iso639.part3)
+        return "+".join(matched_langcodes)
+
+    # try to match 639-2b (part2b)
```

Comment:
639-3, 639-2b, etc are different ISO 639 standards. I added "ISO" to try and clarify this, but if the [tesseract docs](https://tesseract-ocr.github.io/tessdoc/Data-Files-in-different-versions.html) should be included maybe at the `{lang} is not a language supported by Tesseract.` warning, let me know. Otherwise we can include this information on expected inputs into the docs instead.

Reply:
Yeah I think docs are a better place than the return value (easier to update should the link change). I was thinking for eng folks in the comments, but if it's easily discoverable the extra clarification sounds good!